store/mutations.js

```diff
@@ -2,6 +2,17 @@ export default {
   addTask(state, val) {
     state.states.taskList.push(val);
   },
+  deleteTask(state, val) {
+    const list = state.states.taskList;
+
+    for (let i = 0; i < list.length; i++) {
+      if (list[i].id === val.id) {
+        const index = list.indexOf(list[i]);
+
+        list.splice(index, 1);
+      }
+    }
```

Comment:
While imperative way of doing things is fine, it's suggested that we try to code in a declarative way. 
```suggestion
   state.states.taskList = state.states.taskList.filter(task => task.id !== val.id)
```

Reply:
I am also removing the item from the list after filtering it out

Reply:
Resolved